spec/lib/ansible/runner_execution_spec.bats

```diff
@@ -0,0 +1,202 @@
+}
+
+@test "[ansible-runner] runs a playbook with variables in a vault encrypted vars file" {
+  echo -n "vault" >> $VAULT_FILE
+  ANSIBLE_VAULT_PASSWORD_FILE=$VAULT_FILE run exec_ansible_runner_cli hello_world_vault_encrypted_vars_file.yml
+  assert_success
+  assert_output --partial '"msg": "Hello World! vars_file_1=vars_file_1_value, vars_file_2=vars_file_2_value"'
+}
+
+@test "[ansible-runner] runs a playbook using roles from github" {
+  setup_roles_dir $DATA_DIR/hello_world_with_requirements_github/roles
+
+  run exec_ansible_runner_cli hello_world_with_requirements_github/hello_world_with_requirements_github.yml
+  assert_success
+  assert_output --partial '"msg": "Hello World! example_var='\''example var value'\''"'
+}
+
+@test "[ansible-runner] runs a role" {
+  setup_roles_dir $ROLES_DIR $DATA_DIR/hello_world_with_requirements_github/roles/requirements.yml
+
+  run exec_ansible_runner_cli_role manageiq.example
+  assert_success
+  assert_output --partial '"msg": "Hello from manageiq.example role! example_var='\''example var value'\''"'
+}
+
+@test "[ansible-runner] vmware collection" {
+  if [ ! -d /var/lib/manageiq/venv ]; then
+    skip "manageiq venv collections are not present"
+  fi
+
+  run exec_ansible_runner_cli vmware.yml
+  assert_failure # We expect to this to fail due to connecting to an unknown vcenter
+  assert_output --partial '"msg": "Unknown error while connecting to vCenter or ESXi API at vcenter_hostname:443 : [Errno -2] Name or service not known"'
+}
+
+@test "[ansible-runner] aws collection" {
+  if [ ! -d /var/lib/manageiq/venv ]; then
+    skip "manageiq venv collections are not present"
+  fi
+
+  run exec_ansible_runner_cli aws.yml
+  assert_failure # We expect to this to fail due to connecting with bad creds
+  assert_output --partial '"msg": "Failed to describe instances: An error occurred (AuthFailure) when calling the DescribeInstances operation: AWS was not able to validate the provided access credentials"'
+}
+
+################################################################################
+
+exec_ansible_runner() {
+  rails runner "resp = Ansible::Runner.run({}, {}, '$DATA_DIR/$1'); puts resp.human_stdout; exit resp.return_code"
+}
+
+exec_ansible_runner_role() {
+  rails runner "resp = Ansible::Runner.run_role({}, {}, '$1', roles_path: '$ROLES_DIR'); puts resp.human_stdout; exit resp.return_code"
+}
+
+@test "[Ansible::Runner] runs a playbook" {
+  run exec_ansible_runner hello_world.yml
+  assert_success
+  assert_output --partial '"msg": "Hello World!"'
+}
+
+@test "[Ansible::Runner] runs a playbook with variables in a vars file" {
+  run exec_ansible_runner hello_world_vars_file.yml
+  assert_success
+  assert_output --partial '"msg": "Hello World! vars_file_1=vars_file_1_value, vars_file_2=vars_file_2_value"'
+}
+
+@test "[Ansible::Runner] runs a playbook with vault encrypted variables" {
+  skip "requires database access"
```

Comment:
So this is annoying. The way creds work through Ansible::Runner _requires_ an Authentication record in the database, and then the interface expect the Authentication id. I don't like that and IMO, Ansible::Runner should be database agnostic.  It should accept string or perhaps an object that responds_to something, but not a literal ActiveRecord id.

Because of this, on a raw container where we don't have the database, I can't run this test. The same test is above via CLI, so I'm not too worried about it.

I've already written up a change where the interface can accept a String or an object, but I want that in a separate PR.